ts/torch_handler/base_handler.py

```diff
@@ -194,18 +199,50 @@ def handle(self, data, context):
         self.context = context
         metrics = self.context.metrics
 
-        data_preprocess = self.preprocess(data)
-
-        if not self._is_explain():
-            output = self.inference(data_preprocess)
-            output = self.postprocess(output)
+        is_profiler_enabled = os.environ.get("ENABLE_TORCH_PROFILER", None)
+        if is_profiler_enabled:
+            output, self.prof = self.infer_with_profiler(data=data)
         else:
-            output = self.explain_handle(data_preprocess, data)
+            data_preprocess = self.preprocess(data)
+
+            if not self._is_explain():
+                output = self.inference(data_preprocess)
+                output = self.postprocess(output)
+            else:
+                output = self.explain_handle(data_preprocess, data)
 
         stop_time = time.time()
         metrics.add_time('HandlerTime', round((stop_time - start_time) * 1000, 2), None, 'ms')
         return output
 
+    def infer_with_profiler(self, data):
+        # Setting the default profiler arguments to profile cpu usage and record shapes
+        # User can override this argument based on the requirement . Ex: Profiling Gpu
+        if not self.profiler_args:
+            self.profiler_args["activities"] = [ProfilerActivity.CPU]
+            self.profiler_args["record_shapes"] = True
+
+        if "on_trace_ready" not in self.profiler_args:
+            result_path = "/tmp/pytorch_profiler"
```

Comment:
@shrinath-suresh I wonder if we give this option to user to choose the result_path and if not provided we fall back to default path.

Reply:
Also adding/extending one of the examples, like Resnet or BERT on using this profilers would be very helpful to show the usage.

Reply:
@HamidShojanazeri Sure Hamid.. I will pick any one example (mnist or resnet) to showcase profiler integration and update the readme accordingly. Thanks for the suggestion

Reply:
@HamidShojanazeri Regarding the profiler path.. By default the outputs get stored into "/tmp/pytorch_profiler". User can override this behaviour like [this](https://github.com/shrinath-suresh/serve/blob/c5568731c1dcac9859e05375938e811eaab3cb56/test/pytest/profiler_utils/resnet_profiler_override.py#L11). 

@HamidShojanazeri Do you have any suggestions for default path ? At `base_handler.py` i couldn't find any default path (model store?) to store profiler trace files. `/tmp/pytorch_profiler` looks hardcoded.. Any suggestion ?

Reply:
For the profiler path I'd suggest using the same path the pytorch profiler uses by default. If 2 succesive runs of a profile just create new directories in the `/tmp/pytorch_profiler` then there is no need to overwrite but if they do then it's best to allow users to configure what the output path is

Reply:
> I will pick any one example (mnist or resnet) to showcase profiler integration and update the readme accordingly. Thanks for the suggestion

It would be great. would you be able to include it in this PR or it would be a separate one.